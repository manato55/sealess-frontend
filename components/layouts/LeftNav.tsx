import {useState,useEffect} from 'react'
import styled from 'styled-components'
import Link from 'next/link'
import {useRouter} from 'next/router'
import mediaQuery from "styled-media-query";
import {useAuth} from '../../hooks/useAuth'


export const LeftNav = () => {
    const router = useRouter()
    const {logout, me, user}  = useAuth();
    const [currentPathname, setCurrentPathname] = useState<string>('')

    useEffect(() => {
        const initialAction = async() => {
            await me()
        }
        initialAction()
    }, [router])

    useEffect(() => {
        setCurrentPathname(router.pathname)
    }, [router.pathname])

    return (
        <Flex>
            <NavBox>
                {user?.user_type === 2 ?
                    <span>
                        <Link href="/history">
                            <a>
                                {currentPathname.match('/history*') ? 
                                    <CurrentTab>決定済み</CurrentTab>
                                    :<p>決定済み</p>
                                }
                            </a>
                        </Link>
                        <Link href="/returned">
                            <a>
                                {currentPathname.match('/returned*') ? 
                                    <CurrentTab>返却案件</CurrentTab>
                                    :<p>返却案件</p>
                                }
                            </a>
                        </Link>
                        <Link href="/recieve">
                            <a>
                                {currentPathname.match('/recieve*') ? 
                                    <CurrentTab>受信ボックス</CurrentTab>
                                    :<p>受信ボックス</p>
                                }
                            </a>
                        </Link>
                        <Link href="/progress/index/1">
                            <a>
                                {currentPathname.match('/progress/*') ? 
                                    <CurrentTab>未承認案件</CurrentTab>
                                    :<p>未承認案件</p>
                                }
                            </a>
                        </Link>
                        <Link href="/route-tmpl">
                            <a>
                                {currentPathname.match('route-tmpl*') ? 
                                    <CurrentTab>ルート作成</CurrentTab>
                                    :<p>ルート作成</p>
                                }
                            </a>
                        </Link>
                        <Link href="/unreached">
                            <a>
                                {currentPathname.match('unreached*') ? 
                                    <CurrentTab>未着案件</CurrentTab>
                                    :<p>未着案件</p>
                                }
                            </a>
                        </Link>
                        <Link href="/agent">
                            <a>
                                {currentPathname.match('agent*') ? 
                                    <CurrentTab>代理設定</CurrentTab>
                                    :<p>代理設定</p>
                                }
                            </a>
                        </Link>
                    </span>
                :''}
                <a onClick={logout}><p>ログアウト</p></a>
            </NavBox>
        </Flex>
    )
}

const mediaMobile = mediaQuery.lessThan("medium");

const Flex = styled.div`
    ${mediaMobile `
    display: none;
    `}
    flex: 2;
    background: gainsboro;
`;

const NavBox = styled.div`
    text-align: center;

    p {
        margin: 0;
        padding: 20px 0;
    }

    p:hover {
        background: #c0c0c0;
    }

    a {
        text-decoration: none;
        color: black;
        cursor: pointer
    }
`;

const CurrentTab = styled.p`
    background: #1e90ff;
`;

export default LeftNav