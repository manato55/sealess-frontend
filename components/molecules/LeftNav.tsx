import React from 'react'
import styled from 'styled-components'
import {useAuth} from '../../hooks/useAuth'
import { useRecoilValue } from 'recoil'
import { userStatus } from '../../store/atom'
import Label from '../molecules/Label'

export const LeftNav = () => {
    const {logout}  = useAuth();
    const user = useRecoilValue(userStatus)

    return (
        <>
            <NavWrapper>
                {user?.user_type === 2 &&
                    <span>
                        <Label
                            path={'/history'}
                            tagName={'決定済み'}
                        />
                        <Label
                            path={'/returned'}
                            tagName={'返却案件'}
                        />
                        <Label
                            path={'/recieve'}
                            tagName={'受信ボックス'}
                        />
                        <Label
                            path={'/progress/index/1'}
                            tagName={'未承認案件'}
                            pathMatch={'/progress/*'}
                        />
                        <Label
                            path={'/route-tmpl'}
                            tagName={'ルート作成'}
                        />
                        <Label
                            path={'/unreached'}
                            tagName={'未着案件'}
                        />
                        <Label
                            path={'/agent'}
                            tagName={'代理設定'}
                        />
                        <Label
                            path={'/search/1'}
                            tagName={'案件検索'}
                            pathMatch={'/search*'}
                        />
                    </span>
                }
                <a onClick={logout}><p>ログアウト</p></a>
            </NavWrapper>
        </>
    )
}

const NavWrapper = styled.div`
    text-align: center;

    p {
        margin: 0;
        padding: 20px 0;
    }

    p:hover {
        background: #c0c0c0;
    }

    a {
        text-decoration: none;
        color: black;
        cursor: pointer
    }
`;

export default LeftNav