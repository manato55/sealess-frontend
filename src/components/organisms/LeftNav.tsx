import React from 'react';
import styled from 'styled-components';
import { useAuthenticate, useLoginCheck } from '../../hooks/useAuth';
import { useRecoilState, useSetRecoilState } from 'recoil';
import { userStatus, http, searchKeyword } from '../../store/atom';
import Label from '../molecules/Label';
import router from 'next/router';

export const LeftNav = (): React.ReactElement => {
  const [user, setUser] = useRecoilState(userStatus);
  const [keyword, setKeyword] = useRecoilState(searchKeyword);
  const setHttpStatus = useSetRecoilState(http);
  const { logout } = useLoginCheck();

  const onClick = async () => {
    if (!confirm('ログアウトしますか？')) {
      return;
    }
    const res = await logout();
    if (!res.isFailure) {
      setUser(undefined);
      localStorage.removeItem('token');
      // 検索キーワード削除
      setKeyword({ ...keyword, task: null, name: null, year: null });
      router.push('/login');
    } else {
      setHttpStatus(res.error.code);
    }
  };

  return (
    <>
      <NavWrapper>
        {user?.user_type === 2 ? (
          <span>
            <Label path={'/history'} tagName={'決定済み'} />
            <Label path={'/returned'} tagName={'返却案件'} />
            <Label path={'/recieve'} tagName={'受信ボックス'} />
            <Label path={'/progress/index/1'} tagName={'未承認案件'} pathMatch={'/progress/*'} />
            <Label path={'/route-tmpl'} tagName={'ルート作成'} />
            <Label path={'/unreached'} tagName={'未着案件'} />
            <Label path={'/agent'} tagName={'代理設定'} />
            <Label path={'/search/1'} tagName={'案件検索'} pathMatch={'/search*'} />
          </span>
        ) : user?.user_type === 1 ? (
          <span>
            <Label path={'/dep-admin/users'} tagName={'登録者一覧'} />
          </span>
        ) : user?.user_type === 0 ? (
          <span>
            <Label path={'/admin/register-dep-manager'} tagName={'部門管理者登録'} />
            <Label path={'/admin/dep-admin-user'} tagName={'部門管理者一覧'} />
            <Label path={'/admin/dep-register'} tagName={'部門等登録'} />
            <Label path={'/admin/dep-index'} tagName={'部門等一覧'} />
            <Label path={'/admin/all-users'} tagName={'一般登録者一覧'} />
          </span>
        ) : (
          ''
        )}
        <a onClick={() => onClick()}>
          <p>ログアウト</p>
        </a>
      </NavWrapper>
    </>
  );
};

const NavWrapper = styled.div`
  text-align: center;
  height: 100vh;

  p {
    margin: 0;
    padding: 20px 0;
  }

  p:hover {
    background: #c0c0c0;
  }

  a {
    text-decoration: none;
    color: black;
    cursor: pointer;
  }
`;

export default LeftNav;
