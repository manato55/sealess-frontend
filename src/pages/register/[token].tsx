import React, { useEffect, useState, useCallback } from 'react';
import { useRouter } from 'next/router';
import { useAuthenticate } from '../../hooks/useAuthenticate';
import Password from '../../components/molecules/Password';
import Button from '../../components/atoms/Button';
import AuthWrapper from '../../components/atoms/AuthWrapper';

export const Token = (): React.ReactElement => {
  const router = useRouter();
  const { tokenCheck } = useAuthenticate();
  const { officialRegistryForOrdinaryUser } = useAuthenticate();
  const [paramsToken, setParamsToken] = useState<any>(router.query.token);
  const [tokenChecker, setTokenChecker] = useState<boolean>(false);
  const [password, setPassword] = useState<string>('');

  useEffect(() => {
    const initialAction = async () => {
      const res = await tokenCheck(paramsToken);
      if (res) {
        setTokenChecker(true);
      }
    };
    initialAction();
  }, [paramsToken]);

  async function submitRegister(): Promise<void> {
    const data: {
      token: string;
      password: string;
    } = {
      token: paramsToken,
      password: password,
    };
    await officialRegistryForOrdinaryUser(data);
  }

  return (
    <>
      {tokenChecker ? (
        <AuthWrapper>
          <Password password={password} setPassword={setPassword} />
          <Button onClick={() => submitRegister()}>登録</Button>
        </AuthWrapper>
      ) : (
        'Loading...'
      )}
    </>
  );
};

export default Token;
