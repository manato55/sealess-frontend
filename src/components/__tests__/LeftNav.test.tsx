import React from 'react';
import LeftNav from '../organisms/LeftNav';
import { RecoilRoot } from 'recoil';
import { mount, shallow } from 'enzyme';
import 'jsdom-global/register';
import renderer from 'react-test-renderer';


describe('LeftNav test', () => {
  let container;

  test('snapshot', async() => {
    container = await mount(
      <RecoilRoot>
        <LeftNav />
      </RecoilRoot>,
    );
    expect(container).toMatchSnapshot();
  });
});
