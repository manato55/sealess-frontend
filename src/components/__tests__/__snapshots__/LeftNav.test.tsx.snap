// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftNav test snapshot 1`] = `
<RecoilRoot>
  <RecoilRoot_INTERNAL>
    <Batcher
      setNotifyBatcherOfChange={[Function]}
    />
    <LeftNav>
      <styled.div>
        <div
          className="sc-gtsrHT ewiVNj"
        >
          <a
            onClick={[Function]}
          >
            <p>
              ログアウト
            </p>
          </a>
        </div>
      </styled.div>
    </LeftNav>
  </RecoilRoot_INTERNAL>
</RecoilRoot>
`;
